adding 5/21 release notes (#3670)
Azure/AKS: CHANGELOG.md

* After May 31, 2023, Ubuntu 18.04 will reach end of life. AKS will continue to update the host OS from Canonical into the Kubernetes 1.24 VHD images. Customers will not receive daily security updates from Canonical past the end of May, but will be able to consume those through a node image update only.
* Each Kubernetes version is supported for 12 months. After 12 months, the minor version will shift to platform support only. Our new [platform support policy](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions?tabs=azure-cli#platform-support-policy) provides customers with Azure infrastructure support while the cluster is in an n-3 version (where n is the latest supported AKS GA minor version). Platform support does not include anything related to Kubernetes functionality and components, but provides customers with additional support beyond what was previously provided for unsupported versions.
* Unattended Upgrades are disabled on Mariner when running on a NVIDIA GPU enabled VM sizes.
* SecurityPatch OS Servicing channel is not supported on Mariner when running on NVIDIA GPU enabled VM sizes.

### Release notes

* Behavior Changes

>    * Enabled aks-support role to "get" access to these cilium CRD for debugging:ciliumnetworkpolicy, ciliumclusterwidenetworkpolicy,ciliumendpoint ciliumidentity, and ciliumnode for debugging.

Reviewer: Added get permissions for `ciliumnetworkpolicy, ciliumclusterwidenetworkpolicy,ciliumendpoint ciliumidentity, and ciliumnode` api-resources to the `aks-service` ClusterRole to enable support workflows.

# Azure Kubernetes Service Changelog

## Release 2023-05-21

Monitor the release status by regions at [AKS-Release-Tracker](https://releases.aks.azure.com/).

### Announcements

> * Docker container runtime for Windows nodepools has been retired as of May 1, 2023. After docker container runtime is retired, you may remain on existing deployed instances but scaling operations will fail, nodepool creation will fail, and you will be out of support. Follow the detailed steps [in our documentation](https://learn.microsoft.com/azure/aks/learn/quick-windows-container-deploy-cli) to upgrade to containerd. AKS will delete all published windows 2019 docker images and it will block scaling up new nodes or creating new agent pools if windows 2019 docker images are used.  

Reviewer 2: ```suggestion
* Docker container runtime for Windows nodepools has been retired as of May 1, 2023. You may remain on existing deployed instances but scaling operations will fail, nodepool creation will fail, and you will be out of support. Follow the detailed steps [in our documentation](https://learn.microsoft.com/azure/aks/learn/quick-windows-container-deploy-cli) to upgrade to containerd. In alignment with this retirement, AKS has deleted all published windows 2019 docker images.  
```

* After May 31, 2023, Ubuntu 18.04 will reach end of life. AKS will continue to update the host OS from Canonical into the Kubernetes 1.24 VHD images. Customers will not receive daily security updates from Canonical past the end of May, but will be able to consume those through a node image update only.
* Each Kubernetes version is supported for 12 months. After 12 months, the minor version will shift to platform support only. Our new [platform support policy](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions?tabs=azure-cli#platform-support-policy) provides customers with Azure infrastructure support while the cluster is in an n-3 version (where n is the latest supported AKS GA minor version). Platform support does not include anything related to Kubernetes functionality and components, but provides customers with additional support beyond what was previously provided for unsupported versions.

> * AKS is gradually rolling out a change that will rotate the token in the kubeconfig credentials. It shall not incur any impact since kubeconfig has the client certificate. Should you see any issue, retrieve the kubeconfig again with `az aks get-credentials`.

Author: which one?

> * AKS is gradually rolling out a change that will rotate the token in the kubeconfig credentials. It shall not incur any impact since kubeconfig has the client certificate. Should you see any issue, retrieve the kubeconfig again with `az aks get-credentials`.

Author: which one?

> * AKS is gradually rolling out a change that will rotate the token in the kubeconfig credentials. It shall not incur any impact since kubeconfig has the client certificate. Should you see any issue, retrieve the kubeconfig again with `az aks get-credentials`.

Author: no the last announcement was carried over from a previous week

> * AKS is gradually rolling out a change that will rotate the token in the kubeconfig credentials. It shall not incur any impact since kubeconfig has the client certificate. Should you see any issue, retrieve the kubeconfig again with `az aks get-credentials`.

Author: no eta yet

> * AKS is gradually rolling out a change that will rotate the token in the kubeconfig credentials. It shall not incur any impact since kubeconfig has the client certificate. Should you see any issue, retrieve the kubeconfig again with `az aks get-credentials`.

Reviewer: pinging on this internally

> * AKS is gradually rolling out a change that will rotate the token in the kubeconfig credentials. It shall not incur any impact since kubeconfig has the client certificate. Should you see any issue, retrieve the kubeconfig again with `az aks get-credentials`.

Author: removed in latest commit

* Unattended Upgrades are disabled on Mariner when running on a NVIDIA GPU enabled VM sizes.
* SecurityPatch OS Servicing channel is not supported on Mariner when running on NVIDIA GPU enabled VM sizes.

### Release notes

* Behavior Changes

>    * Introduced new field MaintenanceWindow into stable version 2023-05-01 to the MaintenanceConfiguration data model which allows users to configure more flexible schedule with custom UTC offsets.

Author: Do you know where I would find this?

>    * Introduced new field MaintenanceWindow into stable version 2023-05-01 to the MaintenanceConfiguration data model which allows users to configure more flexible schedule with custom UTC offsets.

Author: got rid of it @kaarthis

>    * Introduced new field MaintenanceWindow into stable version 2023-05-01 to the MaintenanceConfiguration data model which allows users to configure more flexible schedule with custom UTC offsets.

Reviewer: here's an example - https://github.com/Azure/azure-rest-api-specs/blob/main/specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2023-04-01/examples/MaintenanceConfigurationsGet.json. this is from 04-01 API though

>    * Introduced new field MaintenanceWindow into stable version 2023-05-01 to the MaintenanceConfiguration data model which allows users to configure more flexible schedule with custom UTC offsets.

Reviewer: if not link to the 05-01 API, can we add a link to the docs change having the CLI example of custom UTC offset

>    * Enabled extra validation for user's custom resources in Istio addon.

Author: both?

>    * The OSM addon no longer cleans-up HPA resources (osm-controller-hpa and osm-injector-hpa) on uninstall of the addon due to a label change. This change ensures these resources are cleaned up.

Reviewer: ```suggestion
   * Fixed a bug where [disabling the Open Service Mesh add-on](https://learn.microsoft.com/azure/aks/open-service-mesh-uninstall-add-on) was leaving behind the HorizontalPodAutoscaler resources `osm-controller-hpa` and `osm-injector-hpa`
```

Reviewer: and this should be moved to bug fixes section

Author: done

>    * Enabled aks-support role to "get" access to these cilium CRD for debugging:ciliumnetworkpolicy, ciliumclusterwidenetworkpolicy,ciliumendpoint ciliumidentity, and ciliumnode for debugging.

Reviewer: what's the name of the role? did we introduce a new Role?

Author: The role is not new. The permissions are.

Reviewer: can we add the name of the role? can't find any role named aks-support

   * During cluster cleanup, we no longer delete the etcd backup storage after the cluster has been stopped for 30 days. Deletion of etcd backup only happens when the cluster is deleted.  

>    * For arm clients that use the location header instead of the async-operation header, return bad request 400 if the async operation failed for a client error rather than 500.

Reviewer: is this relevant for external customer notes?

Author: which one?

Author: These were marked as customer facing

   * Enable the toggle to use ForcePodDrain option in Stop MC operation to give some grace period for the pod to stop before deleting the node. 

* Bug Fixes
   * Fixed bug that will recreate IPv6 SLB backend pools if missing on dual-stack clusters.

>    * Azure CNS released bug fix to generate cni conflist if NCs already exist, as well as implementation for CNS to write Cilium conflist. 

Reviewer: CNI?

>    * Fixed bug to prevent customers from listing secrets in agent nodes. 

Reviewer: can we add more details? was some role changed?

Author: For which one?

Reviewer: what was in the fix? did we change some role and the permissions within it?

Author: yes

Author: added new permissions to the role

Reviewer: what is the name of the role?

Author: aks-support

* Component Updates
   * Decrease default CPU request of Image Cleaner's vulnerability scanner from 1 core to half core which may cause client's scanning take longer time.
   * Update container insights addon to version [3.1.8](https://github.com/microsoft/Docker-Provider/blob/ci_prod/ReleaseNotes.md). 

>    * Upgrade Azure Disk CSI driver to [v1.26.4](https://github.com/kubernetes-sigs/azuredisk-csi-driver/releases/tag/v1.26.4) to fix CVE.

Reviewer: can we add link to the CVE? do we have a issue on AKS repo for this?

Reviewer 3: We don't list specific CVE's

>    * Move to the [latest version](https://github.com/Azure/ClusterConfigurationAgent/releases/tag/v1.11.6-aks) of the extensionmanager image.  

Author: done
